generate_trade_report.py

```
@@ -38,26 +81,46 @@ def m(xs):
         else:cur+=1;dur=max(dur,cur);dd=max(dd,peak-e)
     sd=math.sqrt(sum((x-avg)**2 for x in xs)/(n-1)) if n>1 else 0; down=math.sqrt(sum(min(0,x)**2 for x in xs)/n)
     return {'n':n,'net':sum(xs),'pf':gp/gl if gl else (99 if gp else None),'wr':len(wins)/n,'pay':(sum(wins)/len(wins))/(gl/len(losses)) if wins and losses else None,'exp':avg,'dd':dd,'sh':avg/sd if sd else None,'so':avg/down if down else None,'dur':dur}
-def bs(xs,reps=2000):
+def bs(xs: list[float],reps: int=2000) -> Bootstrap | None:
     if len(xs)<10:return None
     rng=random.Random(SEED+len(xs)); n=len(xs); o=[m([xs[rng.randrange(n)] for _ in range(n)]) for _ in range(reps)]
     def ci(k):
         a=sorted(x[k] for x in o if x[k] is not None); return a[int(.025*len(a))],a[int(.975*len(a))-1]
```

Ruff (0.16.2)

[error] Multiple statements on one line (semicolon)

(E702)